mpsc-framework/src/main/java/pl/wsztajerowski/MPSCFramework.java

```diff
@@ -0,0 +1,105 @@
+package pl.wsztajerowski;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.function.Consumer;
+
+class MPSCFramework<REQ, RES> implements AutoCloseable {
```

Comment:
this should be public I guess. also we need better name like BatchedConsumer, or BatchedExecutor